### PR TITLE
Fix platform provider handling in projec parser

### PR DESCRIPTION
**Pull request by oklinov**

### CODE REVIEW

File: fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public JDKProjectParser(
     private Consumer<BuildPlatformConfig> getBuildPlatformConsumer() {
         return ManagementUtils.managementConsumerWrapper(
                 (BuildPlatformConfig config) -> {
-                    jobBuilder.setPlatform(config.getId());
+                    jobBuilder.setPlatform(config.getId(), "");
```
Comment (judovana) [inline fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java]: why this?

Comment (judovana) [reply fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java]: how so?

File: fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java [hunk 2]
```diff
@@ -157,7 +156,8 @@ public JDKProjectParser(
         private JDKVersion jdkVersion;
         private Product product;
         private Set<BuildProvider> buildProviders;
-        private String platformProvider;
+        private String buildPlatformProvider;
+        private String testPlatformProvider;
         private Platform buildPlatform;
```
Comment (judovana) [inline fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java]: that is cooll...

File: fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java [hunk 3]
```diff
@@ -112,8 +112,7 @@ public JDKProjectParser(
     private Consumer<PlatformConfig> getPlatformsConsumer() {
         return ManagementUtils.managementConsumerWrapper(
                 (PlatformConfig platformConfig) -> {
-                    jobBuilder.setPlatformProvider(platformConfig.getProvider());
-                    jobBuilder.setPlatform(platformConfig.getId());
+                    jobBuilder.setPlatform(platformConfig.getId(), platformConfig.getProvider());
```
Comment (judovana) [inline fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java]: From the scope of code in this patch, I'm nto sure I got the change:                     -jobBuilder.setPlatformProvider(platformConfig.getProvider());
-jobBuilder.setPlatform(platformConfig.getId());

replaced by 

(platformConfig.getId(), platformConfig.getProvider()) Where I can not see hadnling og this tuple

File: fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java [hunk 4]
```diff
@@ -285,15 +285,19 @@ void resetPlatform() {
             throw new RuntimeException("Resetting platform error");
         }
 
-        void setPlatform(final String platformId, final String platformProvider) {
+        void setPlatform(final BuildPlatformConfig buildPlatformConfig) {
+            buildPlatform = platformsMap.get(buildPlatformConfig.getId());
```
Comment (judovana) [inline fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java]: Is this actually correct?
Provider is not part  of id afaik. So platformsMap.get  will return  some  os.arch with set of providers, no? But maybe I mess platform and paltformConfig

Comment (oklinov) [reply fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java]: This applies only to JDKTestProjjects, where build platform does not have a provider

**Comment by judovana** [comment]
> Provider is not part of id afaik. So platformsMap.get will return some os.arch with set of providers, no? But maybe I mess platform and paltformConfig

ping?

**Pull request merged**